src/Http/Http.Extensions/test/RequestDelegateGenerator/Baselines/MapAction_ExplicitBodyParam_ComplexReturn_Snapshot.generated.txt

```diff
@@ -295,11 +304,25 @@ namespace Microsoft.AspNetCore.Http.Generated
             }
             else
             {
-                return httpContext.Response.WriteAsJsonAsync(obj);
+                return WriteJsonResponseAsync(obj, httpContext, jsonTypeInfo);
+            }
+        }
+
+        [UnconditionalSuppressMessage("Trimming", "IL2026:RequiresUnreferencedCode",
+            Justification = "The 'JsonSerializer.IsReflectionEnabledByDefault' feature switch, which is set to false by default for trimmed ASP.NET apps, ensures the JsonSerializer doesn't use Reflection.")]
+        [UnconditionalSuppressMessage("AOT", "IL3050:RequiresDynamicCode", Justification = "See above.")]
+        private static Task WriteJsonResponseAsync<T>(T? value, HttpContext httpContext, JsonTypeInfo<T> jsonTypeInfo)
+        {
+            var runtimeType = value?.GetType();
+            if (runtimeType is null || jsonTypeInfo.Type == runtimeType || jsonTypeInfo.PolymorphismOptions is not null)
+            {
+                return httpContext.Response.WriteAsJsonAsync(value!, jsonTypeInfo);
```

Comment:
```suggestion
                return httpContext.Response.WriteAsJsonAsync(value, jsonTypeInfo);
```

This seems wrong because when `value` is null, we get here.

Maybe the right fix is to take a `JsonTypeInfo<T?> jsonTypeInfo`?

Reply:
Yep, I tweaked the nullability in https://github.com/dotnet/aspnetcore/pull/48573/commits/a9b8d8ba5f49b597ce14f091f39abb7a4e4551ad